avm/res/insights/data-collection-rule/main.bicep

```diff
@@ -73,7 +73,7 @@ var dataCollectionRulePropertiesUnion = union(
         dataSources: dataCollectionRuleProperties.dataSources
       }
     : {},
-  dataCollectionRuleProperties.kind == 'Linux' || dataCollectionRuleProperties.kind == 'Windows' || dataCollectionRuleProperties.kind == 'All' || dataCollectionRuleProperties.kind == 'Direct'
+  dataCollectionRuleProperties.kind == 'Linux' || dataCollectionRuleProperties.kind == 'Windows' || dataCollectionRuleProperties.kind == 'All' || dataCollectionRuleProperties.kind == 'Direct' || dataCollectionRuleProperties.kind == 'WorkspaceTransforms'
```

Comment:
Not strictly related to this PR, but I wonder if 
```bicep
contains(['Linux', 'Windows', 'All', 'Direct', 'WorkspaceTransforms'], dataCollectionRuleProperties.kind)
```
wouldn't scale a bit better